CODE_STYLE.md

```diff
@@ -0,0 +1,943 @@
+  * **test**: Adding missing or correcting existing tests
+  * **chore**: Changes to the build process or auxiliary tools and libraries such as documentation
+    generation
+
+  #### Scope
+  
+  The scope could be anything specifying the place of the commit change. Common examples:
+  
+  * **`<plugin-name>`**: This could be `basic-auth`, or `ldap` for example
+  * **api**: the RESTful management api
+  * **proxy**: the core proxy functionality
+  * **balancer**: the loadbalancer
+  * **dns**: dns resolution
+  * **dao**: datastore persistence functionality
+  * **cache**: for Lua or shm based caching
+  * **cli**: command line interface
+  * **deps**: for updating dependencies
+  * **conf**: for configuration related changes
+  
+  You can use `*` when the change affects more than a single scope.
+
+  ### Subject
+
+  The subject contains succinct description of the change:
+
+  * use the imperative, present tense: "change" not "changed" nor "changes"
+  * don't capitalize first letter
+  * no dot (.) at the end
+
+  ### Body
+
+  Just as in the **subject**, use the imperative, present tense: "change" not "changed" nor "changes".
+  The body should include the motivation for the change and contrast this with previous behavior.
+
+  ### Footer
+
+  The footer should contain any information about **Breaking Changes** and is also the place to
+  [reference GitHub issues that this commit closes](https://help.github.com/articles/closing-issues-via-commit-messages/).
+
+  **Breaking Changes** should start with the word `BREAKING CHANGE:` with a space or two newlines.
+  The rest of the commit message is then used for this.
+
+  A detailed explanation can be found in this [document](https://docs.google.com/document/d/1QrDFcIiPjSLDn3EL15IJygNPiHORgU1_OOAqWjiDU5Y/edit#).
+
+  **[[back to top]](#TOC)**
+
+### <a name='pullrequests'>Pull requests</a>
+
+
+  - First of all, make sure to base your work on the [appropriate branch][#branches]:
+
+    ```
+    # a bugfix branch for `next` would be prefixed by fix/
+    # a bugfix branch for `master` would be prefixed by hotfix/
+    $ git checkout -b feature/my-feature next
+    ```
+
+  - Write insightful and descriptive commit messages, formatted as [specified][#commits].
+
+  - Please **include the appropriate test cases** for your patch.
+
+  - Make sure all tests pass before submitting your changes. See the
+    [Makefile operations](/README.md#makefile-operations).
+
+  - Make sure the linter does not throw any errors: `make lint`.
+
+  - Rebase your commits. It may be that new commits have been introduced on `next`. Rebasing
+    will update your branch with the most recent code and make your changes easier to review:
+
+    ```
+    $ git fetch
+    $ git rebase origin/next
+    ```
+
+  - Push your changes:
+
+    ```
+    $ git push origin -u feature/my-feature
+    ```
+
+  - Open a pull request against the [appropriate branch][#branches].
+
+  - If we suggest changes:
```

Comment:
Is "we" sufficiently clarified elsewhere? If not, perhaps clarify here "If Kong's maintainers suggest changes:"